### PR TITLE
feat: add dealer promotion apis

### PR DESCRIPTION
**Pull request by ufkaya**
BREAKING CHANGE

- [CAR-4653 - Garage Carousel: Edit Page - Slogan, Date and Photo Upload](https://autoricardo.atlassian.net/browse/CAR-4653)

- [x] New APIs to get, create and modify the promotion
- [x] New API to buy a product by defining the start date

### CODE REVIEW

**Review by ralucacril — APPROVED** [review]

File: src/services/product.ts [hunk 1]
```diff
@@ -32,3 +32,28 @@ export const purchaseAndUseProduct = async (
     }
   })
 }
+
+export const purchaseAndUseProductWithStartDate = async (
```
Comment (Averethel) [inline src/services/product.ts:36]: Can we go with different naming? I thought that this is the same product but with an additional `startDate` parameter at a first glance.

How about we:
1. rename `purchaseAndUseProduct` to `purchaseAndUseListingProduct`
2. call this `purchaseAndUseGlobalProduct` or `purchaseAndUseDealerProduct`?

WDYT?

File: src/types/models/dealerPromotion.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+interface Data {
```
Comment (Averethel) [inline src/types/models/dealerPromotion.ts:1]: How about:

```suggestion
interface PromotionContent {
```

File: src/types/models/dealerPromotion.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  description?: string
+  title: string
+}
+
+export interface DealerPromotion {
+  dataDe?: Data
```
Comment (Averethel) [inline src/types/models/dealerPromotion.ts:7]: Is this naming enforced by BE? Can we go with something less generic?

**Review by Averethel — CHANGES_REQUESTED** [review]
A few suggestions from my side. Other than naming in the `product` service everything is optional.
Please also link new services/calls in the readme

File: src/services/product.ts (hunk 1 above)
Comment (ufkaya) [reply src/services/product.ts:36]: ok, it is similar to my first approach -> https://github.com/carforyou/carforyou-api-client-pkg/pull/155/commits/3a3af3c31d1352e476f33798e8f500e85b887823

I will rename them to `purchaseAndUseListingProduct` and `purchaseAndUseDealerProduct`

File: src/types/models/dealerPromotion.ts (hunk 3 above)
Comment (ufkaya) [reply src/types/models/dealerPromotion.ts:7]: Yes, https://dealer-service.preprod.carforyou.ch/swagger-ui/index.html#/Dealer/createPromotionUsingPOST

Comment (Averethel) [reply src/types/models/dealerPromotion.ts:7]: Can you check with them? It should be a low effort to rename a database field and if it makes our code more readable it's definitely worth it.

Comment (ufkaya) [reply src/types/models/dealerPromotion.ts:7]: And how would you rename them?

Comment (masone) [reply src/types/models/dealerPromotion.ts:7]: maybe

```
{ data: {
  de: {},
  fr: {},
 ...
}
```

but I'd do whatever backend is comfortable with. would not change the database if we follow this naming pattern in other places already

Comment (ufkaya) [reply src/types/models/dealerPromotion.ts]: @Averethel I checked with BE, is not possible, they implemented it this way e2e

**Comment by lkappeler** [comment]
@ufkaya Can you add the new endpoints to the readme, thanks

**Review by Averethel — APPROVED** [review]
Looks good to me. Please link the new calls in the readme.
Don't forget that this is a `BREAKING CHANGE` since we renamed the purchase product call

**Pull request merged**